### ISSUE TITLE
Error downloading song (Latin-1)

### ISSUE THREAD

**Issue by davidscherf** [vicwomg/pikaraoke#314]
**Describe the bug**
Search a song you do not have
AEce22afpbU

The song never downloads
Console output says latin-1 codec can't encode character '\uff5c' in position 69: ordinal not in range(256)

**To Reproduce**

1. Search on phone for song
2. Select download an queue
3. See error on console

**Expected behavior**
Download the video and queue it

**Screenshots**
![image](https://github.com/vicwomg/pikaraoke/assets/37095365/4d4ed00d-c0ba-4780-9c6f-09200bff9ddd)

**Platform**
 - OS: Raspberry PI 400
 - OS version: Raspberry PI OS 5.1
 - Browser: chrome
 - Youtube Downloader: 2023.12.30

**Additional context**
Seems to be a bug report from 2022 about this issue here https://github.com/yt-dlp/yt-dlp/issues/5341

**Comment by Hartebee5t** [comment]
I don't how mine makes a difference but I've just tried that same youtube URL (including the AEce22afpbU part) and it downloded and played on my Pi 5 using my Android phone as the browser for the app. The track is a Karafun edition of Under The Bridge.
I know I'm a minor version or two behind the PiKaraoke curve (mine still needs the '-prefer-ip' switch) and there are probably a few updates available on the main OS.

Regards
David

**Comment by vicwomg** [comment]
Yes that is strange because this song does not include any special
characters, usually that's the case with these sorts of issues. Are you
sure it was this song and not some other non-english song?

**Comment by Franck77590** [comment]
Hello, i've the same issue with vietnamese or french songs (like karafun)
If the name of the song have |  (pipe) or * (asterisk) 
![image](https://github.com/user-attachments/assets/c10effcb-f2f5-473c-a5e3-71c6dd40f481)

my log :

ERROR: 'latin-1' codec can't encode character '\uff0a' in position 71: ordinal not in range(256)
[2024-08-03 12:12:28] ERROR: Error code while downloading, retrying once...
ERROR: 'latin-1' codec can't encode character '\uff0a' in position 71: ordinal not in range(256)
[2024-08-03 12:12:37] ERROR: Error downloading song: https://www.youtube.com/watch?v=ZocabPIVreU
[2024-08-03 12:13:08] INFO: Downloading video: https://www.youtube.com/watch?v=AEz6BfE4__U
ERROR: 'latin-1' codec can't encode character '\uff0a' in position 66: ordinal not in range(256)
[2024-08-03 12:13:17] ERROR: Error code while downloading, retrying once...
ERROR: 'latin-1' codec can't encode character '\uff0a' in position 66: ordinal not in range(256)
[2024-08-03 12:13:24] ERROR: Error downloading song: https://www.youtube.com/watch?v=AEz6BfE4__U
[2024-08-03 12:15:21] INFO: Song ending: karaok▒ daniel balavoine le chanteur
[2024-08-03 12:16:31] INFO: Downloading video: https://www.youtube.com/watch?v=AEz6BfE4__U
ERROR: 'latin-1' codec can't encode character '\uff0a' in position 66: ordinal not in range(256)
[2024-08-03 12:16:39] ERROR: Error code while downloading, retrying once...
ERROR: 'latin-1' codec can't encode character '\uff0a' in position 66: ordinal not in range(256)
[2024-08-03 12:16:48] ERROR: Error downloading song: https://www.youtube.com/watch?v=AEz6BfE4__U
[2024-08-03 12:16:57] INFO: Searching YouTube for: karafun karaoke
[2024-08-03 12:17:03] INFO: Downloading video: https://www.youtube.com/watch?v=T3YBQqc9xLg
ERROR: 'latin-1' codec can't encode character '\uff5c' in position 65: ordinal not in range(256)
[2024-08-03 12:17:11] ERROR: Error code while downloading, retrying once...
[2024-08-03 12:17:15] INFO: Searching YouTube for: karafun karaoke
ERROR: 'latin-1' codec can't encode character '\uff5c' in position 65: ordinal not in range(256)
[2024-08-03 12:17:18] ERROR: Error downloading song: https://www.youtube.com/watch?v=T3YBQqc9xLg
[2024-08-03 12:18:35] INFO: Song ending: None

In raspi-config i tried to add/change local zone (with reboot) but i've got the same issue.
![image](https://github.com/user-attachments/assets/7d7268c8-dfdc-42a9-af96-4a4148737eff)
![image](https://github.com/user-attachments/assets/d22142b5-498a-431c-9728-8ce6f9e454a4)

Tested from mobile phone android, ipad mini4 and Pc win11+chrome
i think the problem is from the server not the client.

Do you have an idea ?

my configuration 👍 
raspberry Pi4
![image](https://github.com/user-attachments/assets/28ab9f9e-6735-42b4-96e2-189d2fdb1392)
